core/workflow/loadDPL.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020 CERN and copyright holders of ALICE O².
+ * Author: Ayaan Zaidi <azaidi@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package workflow
+
+import (
+	"sync"
+
+	"github.com/AliceO2Group/Control/common/gera"
+	"github.com/AliceO2Group/Control/core/task"
+)
+
+func LoadDPL(tasks []*task.Class) (workflow Role, err error) {
+	root := new(aggregatorRole)
+
+	for _, taskItem := range tasks {
+		SingleTaskRole := taskRole{
+			roleBase: roleBase{
+				Name:        taskItem.Identifier.Name,
+				parent:      root,
+				Connect:     taskItem.Connect,
+				Constraints: taskItem.Constraints,
+				status: SafeStatus{
+					mu:     sync.RWMutex{},
+					status: 0,
+				},
+				state: SafeState{
+					mu:    sync.RWMutex{},
+					state: 0,
+				},
+				Defaults: gera.MakeStringMapWithMap(taskItem.Defaults.Raw()),
+				Vars:     &gera.StringWrapMap{},
+				UserVars: &gera.StringWrapMap{},
```

Comment:
@teo: which would be the most appropriate field to store the `Class.Properties` StringMap?

Reply:
In `task.Class` it's a `gera.StringMap` (backed by a `map[string]string`). Elsewhere, none.

Please note that `workflow.role*` types and `task.Class` represent different entities. There is no `Properties` field associated with `roleBase` (or any of the `workflow.Role`-related types).

In a `task.Class`, `Properties` means "push these variables as parameters during the `CONFIGURE` transition". The roles tree can define variables, but only the `task.Class` can say "this is the format in which variables `x`, `y` and `z` must be presented to the controlled process".

Reply:
I see, thanks for the clarification.